### PR TITLE
BooleanQuery changes + Added fields to SearchQueryResponse

### PR DESCRIPTION
**Pull request by SecureCake**

### CODE REVIEW

**Comment by chvck** [comment]
As this is a breaking change we can't fix the Buf Breaking but should fix up the lint.

**Review by chvck — CHANGES_REQUESTED** [review]

File: couchbase/search/v1/search.proto [hunk 1]
```diff
@@ -241,7 +240,36 @@ message SearchQueryRequest {
...
+    bytes locations = 5;
+    map<string, bytes> fields = 6;
+    map<string, Fragment> fragments = 7;
+  }
+
+  message Fragment {
+    repeated string content = 1;
+  }
+
+  message Facet {
```
Comment (chvck) [inline couchbase/search/v1/search.proto:258]: This is missing the date_range, numeric_range, term_range fields.

File: couchbase/search/v1/search.proto [hunk 2]
```diff
@@ -241,7 +240,36 @@ message SearchQueryRequest {
...
 
 message SearchQueryResponse {
-  repeated bytes hits = 1;
+
+  message SearchQueryRow {
+    string id = 1;
+    double score = 2;
+    string index = 3;
+    bytes explanation = 4;
+    bytes locations = 5;
```
[the remark this answers is not in the document]
Comment (SecureCake) [reply couchbase/search/v1/search.proto:249]: Looking at the RFC it seems like it's a list? https://github.com/couchbaselabs/sdk-rfcs/blob/master/rfc/0052-sdk3-full-text-search.md#searchrow

File: couchbase/search/v1/search.proto (hunk 1 above)
Comment (SecureCake) [reply couchbase/search/v1/search.proto:258]: I put those in the enum FaceResultType below. I don't remember where I found this though, it's not specified in the RFC strangely. It was probably in the SDK instead https://github.com/couchbaselabs/sdk-rfcs/blob/master/rfc/0052-sdk3-full-text-search.md#searchfacetresult

Comment (chvck) [reply couchbase/search/v1/search.proto:267]: FacetResultType is just an enum? The ranges are lists, see https://docs.couchbase.com/server/current/fts/fts-search-response-facets.html

File: couchbase/search/v1/search.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (chvck) [reply couchbase/search/v1/search.proto]: Happy enough for it to be list that just condenses the map keys (field and term) into the Location type - if that seems like better UX.

File: couchbase/search/v1/search.proto [hunk 3]
```diff
@@ -265,19 +265,36 @@ message SearchQueryResponse {
...
   message Facet {
-    FacetResultType facet_result_type = 1;
     string name = 2;
     string field = 3;
     int64 total = 4;
     int64 missing = 5;
     int64 other = 6;
+    repeated Terms terms = 7;      
+    repeated NumericRange numeric_ranges = 8;
+    repeated DateRange date_ranges = 9;
```
Comment (SecureCake) [inline couchbase/search/v1/search.proto:275]: Protobuf doesn't support keywords (repeated, optional...) inside oneof, so I went with this way instead.

Comment (chvck) [reply couchbase/search/v1/search.proto:275]: I think it'd look like https://github.com/couchbase/protostellar/pull/31/files but FacetResult etc...?

**Review by chvck — APPROVED** [review]

**Pull request merged**